adal/src/main/java/com/microsoft/aad/adal/AuthenticationResult.java

```diff
@@ -139,9 +145,16 @@ static AuthenticationResult createResult(final TokenCacheItem cacheItem) {
             return result;
         }
 
-        return new AuthenticationResult(cacheItem.getAccessToken(), cacheItem.getRefreshToken(),
-                cacheItem.getExpiresOn(), cacheItem.getIsMultiResourceRefreshToken(),
-                cacheItem.getUserInfo(), cacheItem.getTenantId(), cacheItem.getRawIdToken());
+        final AuthenticationResult result =
+                new AuthenticationResult(cacheItem.getAccessToken(), cacheItem.getRefreshToken(),
+                        cacheItem.getExpiresOn(), cacheItem.getIsMultiResourceRefreshToken(),
+                        cacheItem.getUserInfo(), cacheItem.getTenantId(), cacheItem.getRawIdToken(), null);
+
+        if (cacheItem.isExtendedLifetimeValid()) {
```

Comment:
I'm so confused here. If we check isExtendedLifetimeValid, then why not also checking expiresOn? There is a extended time existed but expired is different that it does not exist.